### PR TITLE
feat: add eigenpod bindings

### PR DESCRIPTION
**Pull request by shrimalmadhur**
Fixes # .

### Reviewer Checklist
- [x] Code is well-documented
- [x] Code adheres to Go [naming conventions](https://go.dev/doc/effective_go#names)
- [ ] Code deprecates any old functionality before removing it

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,7 +11,10 @@ help:
 	@grep -E '^[a-zA-Z0-9_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
 bindings: ## generates contract bindings
-	cd contracts && rm -rf bindings/* && ./generate-bindings.sh 
+	cd contracts && rm -rf bindings/* && ./generate-bindings.sh
```
Comment (samlaf) [inline Makefile:14]: think bindings should also generate the eigenpod-bindings, to make sure we don't forget?

File: chainio/clients/eigenpod/reader.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	logger    logging.Logger
+	ethClient eth.HttpBackend
+	*bindings.IEigenPodCaller
+}
+
+type ManagerChainReader struct {
+	logger    logging.Logger
+	ethClient eth.HttpBackend
+	*bindings.IEigenPodManagerCaller
+}
```
Comment (samlaf) [inline chainio/clients/eigenpod/reader.go]: why not merge them in a single struct like we do everything else?
If there's a good reason then add as comment above the structs

File: chainio/clients/eigenpod/reader.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+func NewReader(
+	address common.Address,
+	ethClient eth.HttpBackend,
+	logger logging.Logger,
+) (*ChainReader, error) {
+	pod, err := NewContractCallerBindings(address, ethClient)
+	if err != nil {
+		return nil, err
```
Comment (samlaf) [inline chainio/clients/eigenpod/reader.go]: wrap errors with context everywhere

File: chainio/clients/eigenpod/reader.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+	return &ManagerChainReader{
+		logger:                 logger,
+		ethClient:              ethClient,
+		IEigenPodManagerCaller: manager,
+	}
+}
+
+func NewReader(
+	address common.Address,
```
Comment (samlaf) [inline chainio/clients/eigenpod/reader.go]:
```suggestion
	eigenpodAddress common.Address,
```
just having address is error prone I feel. Or at least add a comment explaining whta address this is

File: chainio/clients/eigenpod/generate.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+cleanup_bindings_dir() {
+  echo "Cleaning up the bindings directory"
+  rm -rf ${BINDINGS_DIR}
+}
+
+create_tmp_dir() {
+  echo "Creating a temporary directory"
+  mkdir -p ${TMP_DIR}
+}
```
Comment (samlaf) [inline chainio/clients/eigenpod/generate.sh]: bash has a function for this which is pretty cool, also no risk of poisoning your local directory.
```
local tmpdir=$(mktemp -t mydir)
```
this creates a directory in `/tmp`

File: chainio/clients/eigenpod/generate.sh
Comment (samlaf) [inline chainio/clients/eigenpod/generate.sh:1]: I would add a comment somewhere, whether in a chainio/clients/eigenpod/README, or in this file, or in the makefile, explaining why we generate bindings separately, and we don't use the submodule that we use for everything else, etc.

File: chainio/clients/eigenpod/reader.go (hunk 2 above)
Comment (shrimalmadhur) [reply chainio/clients/eigenpod/reader.go]: yea there is. most people who want to manage an eigenpod only need the first one. Eigenpod manager is mostly for creating a new eigenpod. I will add that comment.

File: Makefile (hunk 1 above)
Comment (shrimalmadhur) [reply Makefile:14]: it does generate https://github.com/Layr-Labs/eigensdk-go/blob/7c76ea7cdb3bc3c0753de009c01054a8c09446d3/contracts/generate-bindings.sh#L59

Comment (shrimalmadhur) [reply Makefile:14]: c68b91180062786b36dc2566fb7dc393ac673d89

File: chainio/clients/eigenpod/reader.go (hunk 4 above)
Comment (shrimalmadhur) [reply chainio/clients/eigenpod/reader.go]: c68b91180062786b36dc2566fb7dc393ac673d89

File: chainio/clients/eigenpod/generate.sh
Comment (shrimalmadhur) [reply chainio/clients/eigenpod/generate.sh:1]: c68b91180062786b36dc2566fb7dc393ac673d89

File: chainio/clients/eigenpod/reader.go (hunk 2 above)
Comment (shrimalmadhur) [reply chainio/clients/eigenpod/reader.go]: c68b91180062786b36dc2566fb7dc393ac673d89

File: chainio/clients/eigenpod/reader.go (hunk 3 above)
Comment (shrimalmadhur) [reply chainio/clients/eigenpod/reader.go]: c68b91180062786b36dc2566fb7dc393ac673d89

**Pull request merged**